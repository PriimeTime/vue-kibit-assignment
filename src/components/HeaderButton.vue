<script setup lang="ts">
const emit = defineEmits(["click"]);
</script>

<template>
  <button
    @click="emit('click')"
    class="p-2 w-20 mx-2 rounded-md text-white border-teal-900 border-2 bg-teal-900 flex justify-center items-center whitespace-nowrap"
  >
    <slot />
  </button>
</template>
