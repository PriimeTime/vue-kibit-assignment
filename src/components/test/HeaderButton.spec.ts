import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import HeaderButton from "@/components/HeaderButton.vue";

describe("HeaderButton", () => {
  it("renders without crashing", () => {
    const wrapper = mount(HeaderButton);
    expect(wrapper.exists()).toBe(true);
  });

  it("emits 'click' event when clicked", async () => {
    const wrapper = mount(HeaderButton);
    await wrapper.trigger("click");
    expect(wrapper.emitted("click")).toBeTruthy();
  });

  it("matches snapshot", () => {
    const wrapper = mount(HeaderButton, {
      slots: {
        default: "Click Me",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
