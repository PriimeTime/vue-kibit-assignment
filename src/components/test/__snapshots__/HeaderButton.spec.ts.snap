// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeaderButton > matches snapshot 1`] = `"<button class="p-2 w-20 lg:mx-2 rounded-md text-white text-xs md:text-xl border-teal-900 border-2 bg-teal-900 flex justify-center items-center whitespace-nowrap">Click Me</button>"`;
